mockwebserver/src/main/java/okhttp3/mockwebserver/MockWebServer.java

```diff
@@ -681,7 +681,6 @@ private void handleWebSocketUpgrade(Socket socket, BufferedSource source, Buffer
     } catch (IOException e) {
       webSocket.failWebSocket(e, null);
     } finally {
-      closeQuietly(sink);
```

Comment:
Above on line 671 we’re handing off the sink to another thread, so this operation was unsafe.